superficial/src/main/scala/superficial/TwoComplex.scala

```diff
@@ -403,6 +403,42 @@ trait TwoComplex { twoComplex =>
     }
     newComplex
   }
+
+  //Finds neighbours of a vertex
+  def VertexNbr(v: Vertex): Set[Vertex] = {
+    val s = (twoComplex.edges.filter(_.initial == v).map(_.terminal)).union(twoComplex.edges.filter(_.terminal == v).map(_.initial))
+    s.union(Set(v))
+  }
+
+  //Collects a first order neighbourhood of a set onto a set
+  def SetNbr(s: Set[Vertex]): Set[Vertex] = {
+    //Accumulates first order neighbours onto a list
+    def ListNbr(s: List[Vertex], accum: List[Vertex]): List[Vertex] = {
+      s match {
+        case x::rest => ListNbr(rest, accum ++ VertexNbr(x))
+        case Nil => accum
+      }
+    
+    }
+    ListNbr(s.toList, List[Vertex]()).toSet
+  }
```

Comment:
* This can be simplified to `s.flatMap(VertexNbr(_))`. 
* In any case storing intermediate lists which will have duplicates is inefficient (i.e., `accum` should be a set if one does this using recursion, though one should just flatMap).

Reply:
Incidentally, a third way of doing things which is more general than `flatMap` is to use collection operations:
```scala
def SetNbr(s: Set[Vertex]): Set[Vertex] = s.map(VertexNbr).fold(Set())(_ union _)
```

Reply:
Makes sense. I guess I wasn't used to using flatMap and hence didn't even consider using it. Change implemented.